### PR TITLE
implement bike distance calculations and sorting

### PR DESCRIPTION
**Pull request by evanm510**

### CODE REVIEW

**Review by campbeb3 — APPROVED** [review]
Looks good!

File: bike_kollective/lib/data/model/bike.dart [hunk 1]
```diff
@@ -20,6 +19,7 @@ class BikeModel {
   final DateTime locationUpdated;
   final List<BKDocumentReference> rides;
   final List<BKDocumentReference> issues;
+  final double? distance; // Optional distance property
```
Comment (kenneth-tucker) [inline bike_kollective/lib/data/model/bike.dart]: Distance should be derived from the active user's location and the location? This assumes only one user, which isn't true

Comment (kenneth-tucker) [reply bike_kollective/lib/data/model/bike.dart]: I recommend removing this from the model object and calculating it elsewhere or creating another class BikeWithDistance that is a tuple of a bike and a distance. The model class should only contain data that is intended for the database.

File: bike_kollective/lib/data/model/bike.dart [hunk 2]
```diff
@@ -95,35 +95,35 @@ class BikeModel {
...
+      BikeType? type,
+      String? description,
+      String? code,
+      String? imageUrl,
+      BikeStatus? status,
+      BKGeoPoint? locationPoint,
+      DateTime? locationUpdated,
+      List<BKDocumentReference>? rides,
+      List<BKDocumentReference>? issues,
+      double? distance}) {
```
Comment (kenneth-tucker) [inline bike_kollective/lib/data/model/bike.dart]: Undo changes to this file

File: bike_kollective/lib/ui/explore_bikes/explore_bikes_screen.dart [hunk 3]
```diff
@@ -19,13 +24,60 @@ class ExploreBikesScreenState extends ConsumerState<ExploreBikesScreen> {
...
+          child: BikesViewer(
+            availableBikes: bikesWithDistances,
+            isMyBikes: false,
+          ),
+        );
+      },
     );
   }
+
+  double _calculateDistance(BKGeoPoint userLocation, BKGeoPoint bikeLocation) {
```
Comment (kenneth-tucker) [inline bike_kollective/lib/ui/explore_bikes/explore_bikes_screen.dart]: Consider moving this to its own file, in a utils/ directory. Then we can call this where we need the distance, it's ok to run this multiple times

File: bike_kollective/lib/ui/my_bikes_screen/my_bikes_screen.dart [hunk 4]
```diff
@@ -21,13 +25,63 @@ class MyBikesScreenState extends ConsumerState<MyBikesScreen> {
...
+          child: BikesViewer(
+            availableBikes: bikesWithDistances,
+            isMyBikes: true,
+          ),
+        );
+      },
     );
   }
+
+  double _calculateDistance(BKGeoPoint userLocation, BKGeoPoint bikeLocation) {
```
Comment (kenneth-tucker) [inline bike_kollective/lib/ui/my_bikes_screen/my_bikes_screen.dart]: Remove duplicate (import and use a common /utils function)

File: bike_kollective/lib/bike_details_screen.dart [hunk 5]
```diff
@@ -3,10 +3,11 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:bike_kollective/data/model/bike.dart';
 import 'bike_details_view.dart';
 import 'report_issue_dialog.dart';
+import 'package:bike_kollective/data/model/bike_with_distance.dart';
 
 // Define the BikeDetailsScreen widget
 class BikeDetailsScreen extends ConsumerWidget {
-  final BikeModel bike;
+  final BikeWithDistanceModel bike;
```
Comment (kenneth-tucker) [inline bike_kollective/lib/bike_details_screen.dart]: Let's change this to "BikeWithDistance" that is simply a class with two members, the BikeModel and the distance.

**Review by kenneth-tucker — APPROVED** [review]

**Pull request merged**